tests/integration/request-authorizer/src/authorizer.js

```diff
@@ -0,0 +1,38 @@
+function generatePolicy(principalId, effect, resource, context) {
+  const authResponse = {
+    context,
+    principalId,
+  }
+
+  if (effect && resource) {
+    const policyDocument = {
+      Statement: [
+        {
+          Action: 'execute-api:Invoke',
+          Effect: effect,
+          Resource: resource,
+        },
+      ],
+      Version: '2012-10-17',
+    }
+
+    authResponse.policyDocument = policyDocument
+  }
+  return authResponse
+}
+
+export async function authorizerFunction(event, context, callback) {
```

Comment:
this handler is `async` as well as using a `callback`. probably better to use one or the other, not both, to avoid confusion.